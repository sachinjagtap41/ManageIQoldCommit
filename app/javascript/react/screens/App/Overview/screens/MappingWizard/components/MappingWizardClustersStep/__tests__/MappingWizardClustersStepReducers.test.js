import Immutable from 'seamless-immutable';

import clustersReducer from '../MappingWizardClustersStepReducer';
import {
  FETCH_V2V_SOURCE_CLUSTERS,
  FETCH_V2V_TARGET_CLUSTERS
} from '../MappingWizardClustersStepConstants';

import {
  sourceClusters,
  targetClusters
} from '../mappingWizardClustersStep.fixtures';

const initialState = Immutable({
  sourceClusters: [],
  isFetchingSourceClusters: false,
  isRejectedSourceClusters: false,
  errorSourceClusters: null,
  targetClusters: [],
  isFetchingTargetClusters: false,
  isRejectedTargetClusters: false,
  errorTargetClusters: null
});

test('sets default state', () => {
  const action = { type: '@@INIT' };
  const state = clustersReducer(undefined, action);

  expect(state).toMatchSnapshot();
});

describe('fetching source clusters', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_CLUSTERS}_PENDING`
    };
    const prevState = initialState.set('isRejectedSourceClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_SOURCE_CLUSTERS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingSourceClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  describe('is successful,', () => {
    let prevState;
    beforeEach(() => {
      prevState = initialState
        .set('isRejectedSourceClusters', true)
        .set('isFetchingSourceClusters', true);
    });

    test('and there is data', () => {
      const { resources } = sourceClusters;
      const payload = {
        data: {
          ...sourceClusters,
          resources: [
            {
              ...resources
            }
          ]
        }
      };
      const action = {
        type: `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });

    test('and there is no data', () => {
      const payload = {
        data: {
          ...sourceClusters,
          resources: []
        }
      };
      const action = {
        type: `${FETCH_V2V_SOURCE_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });
  });
});

describe('fetching target clusters', () => {
  test('is pending', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_CLUSTERS}_PENDING`
    };
    const prevState = initialState.set('isRejectedTargetClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  test('is rejected', () => {
    const action = {
      type: `${FETCH_V2V_TARGET_CLUSTERS}_REJECTED`,
      payload: 'error'
    };
    const prevState = initialState.set('isFetchingTargetClusters', true);
    const state = clustersReducer(prevState, action);

    expect(state).toMatchSnapshot();
  });

  describe('is successful,', () => {
    let prevState;
    beforeEach(() => {
      prevState = initialState
        .set('isRejectedTargetClusters', true)
        .set('isFetchingTargetClusters', true);
    });
    test('and there is data', () => {
      const { resources } = targetClusters;
      const payload = {
        data: {
          ...targetClusters,
          resources: [
            {
              ...resources
            }
          ]
        }
      };
      const action = {
        type: `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });

    test('and there is no data', () => {
      const payload = {
        data: {
          ...targetClusters,
          resources: []
        }
      };
      const action = {
        type: `${FETCH_V2V_TARGET_CLUSTERS}_FULFILLED`,
        payload
      };
      const state = clustersReducer(prevState, action);

      expect(state).toMatchSnapshot();
    });
  });
});
