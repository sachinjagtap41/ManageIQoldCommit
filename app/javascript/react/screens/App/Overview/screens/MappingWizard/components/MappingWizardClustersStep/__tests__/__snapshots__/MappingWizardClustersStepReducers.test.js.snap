// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetching source clusters is pending 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": true,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is rejected 1`] = `
Object {
  "errorSourceClusters": "error",
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": true,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is successful, and there is data 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [
    Object {
      "0": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000001",
        "id": "1",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster1",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
      "1": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000002",
        "id": "2",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster2",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
      "2": Object {
        "created_on": "2016-07-15T18:28:16Z",
        "drs_automation_level": null,
        "drs_enabled": null,
        "drs_migration_threshold": null,
        "effective_cpu": null,
        "effective_memory": null,
        "ems_id": "10000000000012",
        "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
        "ext_management_system": Object {
          "emstype": "vmwarews",
        },
        "ha_admit_control": null,
        "ha_enabled": null,
        "ha_max_failures": null,
        "href": "http://localhost:3000/api/clusters/10000000000003",
        "id": "3",
        "last_perf_capture_on": null,
        "name": "VMWare Cluster3",
        "type": null,
        "uid_ems": "00000002-0002-0002-0002-00000000024b",
        "updated_on": "2016-07-15T18:28:16Z",
        "v_parent_datacenter": "Datacenter",
      },
    },
  ],
  "targetClusters": Array [],
}
`;

exports[`fetching source clusters is successful, and there is no data 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching target clusters is pending 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": true,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching target clusters is rejected 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": "error",
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": true,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`fetching target clusters is successful, and there is data 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [
    Object {
      "0": Object {
        "created_on": "2016-08-03T02:19:35Z",
        "drs_automation_level": "fullyAutomated",
        "drs_enabled": false,
        "drs_migration_threshold": 3,
        "effective_cpu": 109356,
        "effective_memory": 789977235456,
        "ems_id": "10000000000007",
        "ems_ref": "domain-c7",
        "ems_ref_obj": "domain-c7",
        "ext_management_system": Object {
          "emstype": "rhevm",
        },
        "ha_admit_control": true,
        "ha_enabled": false,
        "ha_max_failures": 1,
        "href": "http://localhost:3000/api/clusters/10000000000006",
        "id": "1",
        "last_perf_capture_on": null,
        "name": "RHV Cluster1",
        "type": null,
        "uid_ems": "domain-c7",
        "updated_on": "2017-02-26T02:33:43Z",
        "v_parent_datacenter": "Default",
      },
      "1": Object {
        "created_on": "2016-08-03T02:19:35Z",
        "drs_automation_level": "fullyAutomated",
        "drs_enabled": false,
        "drs_migration_threshold": 3,
        "effective_cpu": 109356,
        "effective_memory": 789977235456,
        "ems_id": "10000000000008",
        "ems_ref": "domain-c7",
        "ems_ref_obj": "domain-c7",
        "ext_management_system": Object {
          "emstype": "rhevm",
        },
        "ha_admit_control": true,
        "ha_enabled": false,
        "ha_max_failures": 1,
        "href": "http://localhost:3000/api/clusters/10000000000007",
        "id": "2",
        "last_perf_capture_on": null,
        "name": "RHV Cluster2",
        "type": null,
        "uid_ems": "domain-c7",
        "updated_on": "2017-02-26T02:33:43Z",
        "v_parent_datacenter": "Default",
      },
    },
  ],
}
`;

exports[`fetching target clusters is successful, and there is no data 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;

exports[`sets default state 1`] = `
Object {
  "errorSourceClusters": null,
  "errorTargetClusters": null,
  "isFetchingSourceClusters": false,
  "isFetchingTargetClusters": false,
  "isRejectedSourceClusters": false,
  "isRejectedTargetClusters": false,
  "sourceClusters": Array [],
  "targetClusters": Array [],
}
`;
